tools/ports/__init__.py

```diff
@@ -129,16 +129,24 @@ def build_port(src_dir, output_path, build_dir, includes=[], flags=[], exclude_f
       if not os.path.exists(build_dir):
         os.makedirs(build_dir)
       build_dir = src_dir
-    commands = []
-    objects = []
-    for src in srcs:
-      relpath = os.path.relpath(src, src_dir)
-      obj = os.path.join(build_dir, relpath) + '.o'
-      commands.append([shared.EMCC, '-c', src, '-o', obj] + cflags)
-      objects.append(obj)
-
-    system_libs.run_build_commands(commands)
-    system_libs.create_lib(output_path, objects)
+
+    if system_libs.USE_NINJA:
+      ninja_file = os.path.join(build_dir, 'build.ninja')
+      system_libs.ensure_sysroot()
+      system_libs.create_ninja_file(srcs, ninja_file, output_path, cflags=cflags)
+      shared.check_call(['ninja', '-C', build_dir], env=system_libs.clean_env())
```

Comment:
Does it make sense to conditionally add -v for ninja as you did below?

I guess this is preexisting but it seems a little unfortunate that this is duplicated.

Reply:
Done.